src/jit/lsra.cpp

```diff
@@ -5706,6 +5706,166 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
+        {
+            return false;
+        }
+
+        // We don't prefer to spill a register if the weight of recentAssignedRef > weight
+        // of the spill candidate found so far.  We would consider spilling a greater weight
+        // ref position only if the refPosition being allocated must need a reg.
+        unsigned weight1 = BB_ZERO_WEIGHT;
+        unsigned weight2 = BB_ZERO_WEIGHT;
+        if (recentAssignedRef != nullptr)
+            weight1 = getWeight(recentAssignedRef);
+        if (anotherRecentAssignedRef != nullptr)
+            weight2 = getWeight(anotherRecentAssignedRef);
+
+        *recentAssignedRefWeight = (weight1 > weight2) ? weight1 : weight2;
+
+        if (*recentAssignedRefWeight > farthestRefPosWeight)
+        {
+            return false;
+        }
+    }
+    else
+    {
+        if (anotherRecentAssignedRef != nullptr)
+        {
+            // Case 3: recentAssignedRef ==nullptr && anotherRecentAssignedRef != nullptr
+            if (anotherRecentAssignedRef->nodeLocation == refLocation)
+            {
+                // We can't spill a register that's being used at the current location
+                return false;
+            }
+
+            // If the current position has the candidate register marked to be delayed,
+            // check if the previous location is using this register, if that's the case we have to skip
+            // since we can't spill this register.
+            if (anotherRecentAssignedRef->delayRegFree && (refLocation == anotherRecentAssignedRef->nodeLocation + 1))
+            {
+                return false;
+            }
+
+            // We don't prefer to spill a register if the weight of recentAssignedRef > weight
+            // of the spill candidate found so far.  We would consider spilling a greater weight
+            // ref position only if the refPosition being allocated must need a reg.
+            *recentAssignedRefWeight = getWeight(anotherRecentAssignedRef);
+            if (*recentAssignedRefWeight > farthestRefPosWeight)
+            {
+                return false;
+            }
+        }
+        else
+        {
+            // Case 4: recentAssignedRef ==nullptr && anotherRecentAssignedRef == nullptr
+            // fallthrough
+        }
+    }
+    return true;
+}
+
+//----------------------------------------------------------------------------
+// checkReferenceAt: Test assertion conditions for recent assigned RefPosition
+//
+// Arguments:
+//    recentAssignedRef - RefPosition to be tested
+//    refLocation       - LsraLocation of interval being allocated
+//
+// Return Value:
+//    N/A
+//
+// Note: This helper is designed to be used only from allocateBusyReg()
+//
+void LinearScan::checkReferenceAt(RefPosition* recentAssignedRef, LsraLocation refLocation)
+{
+    // The assigned interval has a reference at this location - otherwise, we would have found
+    // this in tryAllocateFreeReg().
```

Comment:
Because I couldn't make a clear statement, I just removed this line from a newly introduced helper `bool LinearScan::checkActiveInterval(Interval* interval, LsraLocation refLocation)`.
https://github.com/dotnet/coreclr/pull/13145/files#diff-93fb9832a03a9783e2a415b581fcb45cR5806

Can you suggested a comment if necessary ?